lib/internal/streams/pipeline.js

```diff
@@ -147,6 +150,35 @@ async function pump(iterable, writable, finish, { end }) {
   }
 }
 
+async function pumpToWeb(readable, writable, finish, { end }) {
+  if (isTransformStream(writable)) {
+    writable = writable.writable;
+  }
+  // https://streams.spec.whatwg.org/#example-manual-write-with-backpressure
+  const writer = writable.getWriter();
+  try {
+    for await (const chunk of readable) {
+      await writer.ready;
+      writer.write(chunk).catch(() => {});
+    }
+
+    await writer.ready;
```

Comment:
Why is this waiting on `writer.ready` here?

Reply:
same as https://github.com/nodejs/node/pull/46307#discussion_r1090012902 for the after iter is completed and there was some error at the last write?